src/sandbox/components/GridLayout.svelte

```diff
@@ -119,19 +127,70 @@
       sliceBlock(transpose ?
         [
           ['#', ...secondaryKeys],
-          ...primaryKeys.map(k1 => [ k1, ...secondaryKeys.map(k2 => expandable(expanded.get(k1)) ? expanded.get(k1).get(k2) : expanded.get(k1)) ])
+          ...primaryKeys.map(k1 => [ k1, ...secondaryKeys.map(k2 => expandable(expanded.get(k1))
+                  ? expanded.get(k1).get(k2) : expanded.get(k1)) ])
         ] :
         [
           ['#', ...primaryKeys],
-          ...secondaryKeys.map(k2 => [ k2, ...primaryKeys.map(k1 => expandable(expanded.get(k1)) ? expanded.get(k1).get(k2) : expanded.get(k1)) ])
+          ...secondaryKeys.map(k2 => [ k2, ...primaryKeys.map(k1 => expandable(expanded.get(k1))
+                  ? expanded.get(k1).get(k2) : expanded.get(k1)) ])
         ]
       );
     }
     return acc;
   }, []);
 
+  function showContent(row, col) {
+    const block = blocks.get(query(blocks).getAt(row, col));
+    const value = block.node ? `${block.node} = ${block.formula}` : block.formula;
+    return value ? value : ""
+  }
+
 </script>
 
+<div
+  class="gridlayout__container gridlines"
+  style={`width: ${nCols * columnWidth}px; height: ${nRows * rowHeight}px;`}
+>
+  {#each tiles as { id, pos: [row, col, rowSpan, colSpan], data, component, node, header } (id)}
+    <div
+      class:node={node}
+      class={ data || component ? 'gridlayout__tile' : 'gridlayout__block-bg' }
+      style={`
+        transform: translate(${col * columnWidth}px, ${row * rowHeight}px);
+        height: ${rowSpan * rowHeight}px;
+        width: ${(colSpan || 1) * columnWidth}px;
+        overflow: ${colSpan ? 'hidden' : 'visible'};
+      `}
+    >
+    {#if component}
+      <CustomRender
+        component={component}
+        rowSpan={rowSpan}
+        colSpan={colSpan}
+        rowHeight={rowHeight}
+        columnWidth={columnWidth}
+        on:resize={({ detail: { nRows: newHeight, nCols: newWidth } }) =>
+          makeSpace(thisSheet, parseInt(id), { newHeight, newWidth })}
+      />
+    {:else if data}
+      {#each data as cell}
+        <div
+          class:bg-white={!transparent}
+          class:dark:bg-dark-500={!transparent}
+          class:bg-transparent={transparent}
+          class:header={header}
+          class="gridlayout__cell gridlines"
+          title={showContent(row, col)}
```

Comment:
This is ugly as hell. Can you even think of the execution complexity here ??? It's at least smth like n^4 and probably worse.
I believe on a sheet with a million cells this is gonna die a hard death.